lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -326,6 +330,23 @@ instance ToText SortOrder where
 instance FromText SortOrder where
     fromText = fromTextToBoundedEnum SnakeLowerCase
 
+defaultTxSortOrder :: SortOrder
+defaultTxSortOrder =  Descending
```

Comment:
Just wondering, why is it better to have `Descending` as a default?

Reply:
Hm indeed… this is currently not used from the API as a "defaultTxSortOrder".

It is currently used for:
1. Ordering the DB read for pending transactions when reading a wallet checkpoint. This appears **pointless** as it will be put in a `Set (Tx, TxMeta)` later. https://github.com/input-output-hk/cardano-wallet/blob/315fec4d4cb56114e1fd59eedd420ea963e3cf29/lib/core/src/Cardano/Wallet/DB/Sqlite.hs#L312
2. `DBSpec` which just needs an arbitrary sort-order.
3. `SqliteFileModeSpec` which just needs an arbitrary sort-order https://github.com/input-output-hk/cardano-wallet/blob/315fec4d4cb56114e1fd59eedd420ea963e3cf29/lib/core/test/unit/Cardano/Wallet/DB/SqliteFileModeSpec.hs#L137
4. `listTransactions` in `WalletLayer`, which is temporary

Maybe we should remove this from `Primitive.Types`.

Reply:
#592

Reply:
If this `Descending` value is used as a default _somewhere_ (be it in the API or elsewhere), then I'd much rather see it defined as a named constant than as a magic value embedded in the functions that use it.

See: https://github.com/input-output-hk/cardano-wallet/pull/592#pullrequestreview-268762445

Reply:
> listTransactions in WalletLayer, which is temporary

What do you mean "temporary" :sweat_smile: ?

---

I think the rationale is: the ordering should come from the caller of the wallet layer. So, starting from the `WalletLayer`, the sort order shouldn't be `Maybe` but a plain value (because a default at this stage would be totally arbitrary). 
At the API-level however, we decide of a default ordering for which `Descending` makes sense (most recent results first).

Reply:
`listTransactions` used a hard-coded `Descending` order instead of its argument.

With a default value in `listTransactions` we do need to refer to `Descending`, which I didn't think about.

>  the sort order shouldn't be Maybe but a plain value (because a default at this stage would be totally arbitrary).

Yes. Unless one argues that defaulting to `Descending` shouldn't be unique to the API. Since you seem positive, I think I'll go with it though.